hplot/label.go

```diff
@@ -0,0 +1,94 @@
+package hplot
+
+import (
+	"gonum.org/v1/plot/plotter"
+	"gonum.org/v1/plot/vg/draw"
+)
+
+type Label struct {
+	Text  string         // Text of the label
+	X, Y  float64        // Position of the label
+	Style draw.TextStyle // Style of the label
+
+	// Compute the position wrt canvas size. This
+	// feature would need to be implemented into the plotter.
+	// (There is an example in gonum/v1/plot/plotter).
+	CanPos bool
+}
+
+// NewLabel creates a new value of type Label.
+func NewLabel(x, y float64, txt string, opts ...LabelOption) Label {
```

Comment:
as is, I am not completely sure this function really adds value.
perhaps just remove it? (together w/ the LabelOption types and func-opts)